packages/ocp-build/ocp-build.1.99.20-beta/opam

```diff
@@ -0,0 +1,55 @@
+(**************************************************************)
+(*                                                            *)
+(*      This file is managed by ocp-autoconf                  *)
+(*  Remove it from `manage_files` in 'ocp-autoconf.config'    *)
+(*  if you want to modify it manually (or use 'opam.trailer') *)
+(*                                                            *)
+(**************************************************************)
+
+opam-version: "1.2"
+maintainer: "Fabrice Le Fessant <fabrice.le_fessant@ocamlpro.com>"
+authors: [
+  "Fabrice Le Fessant <fabrice.le_fessant@ocamlpro.com>"
+]
+homepage: "http://www.typerex.org/ocp-build.html"
+dev-repo: "https://github.com/OCamlPro/ocp-build.git"
+bug-reports: "https://github.com/OCamlPro/ocp-build/issues"
+build: [
+  [     "./configure"
+    "--prefix"
+    "%{prefix}%"
+  ]
+  [ make ]
+]
+install: [
+  [ make "install" ]
+]
+remove: [
+  [ "rm" "-f" "%{prefix}%/bin/ocp-build"   ]
+  [ "rm" "-f" "%{prefix}%/bin/ocp-pp"   ]
+  [ "rm" "-rf" "%{prefix}%/lib/ocaml/typerex/ocp-build"   ]
+  [ "rm" "-rf" "%{prefix}%/lib/ocaml/typerex/ocplib-compat"   ]
+  [ "rm" "-rf" "%{prefix}%/lib/ocaml/typerex/ocplib-debug"   ]
+  [ "rm" "-rf" "%{prefix}%/lib/ocaml/typerex/ocplib-lang"   ]
+  [ "rm" "-rf" "%{prefix}%/lib/ocaml/typerex/ocplib-subcmd"   ]
+  [ "rm" "-rf" "%{prefix}%/lib/ocaml/typerex/ocplib-system"   ]
+  [ "rm" "-rf" "%{prefix}%/lib/ocaml/typerex/ocplib-unix"   ]
+  [ "rm" "-rf" "%{prefix}%/lib/ocaml/typerex/installed.ocp"   ]
+  [ "rm" "-rf" "%{prefix}%/lib/ocaml/site-ocp2/ocp-build"   ]
+  [ "sh" "-exc" "rmdir %{prefix}%/lib/ocaml/typerex || true"   ]
+]
+available: [ocaml-version >= "4.02.1"]
+
+(**************************************************************)
+(*                                                            *)
+(* From opam.trailer:                                         *)
+(*                                                            *)
+(**************************************************************)
+
+conflicts: [ "typerex"  {< "1.99.7"} "typerex-build" ]
+depends: [
+    "ocamlfind" (* not strictly needed by ocp-build, but
```

Comment:
Could you add the ```build``` attribute to ocamlfind ?

Reply:
`ocamlbuild` is not a build dependency, but an execution dependency. So, no...

Reply:
The semantics of the ```build``` attribute it that it rebuilds the package and all its dependencies in case of an update. As soon as you do not depend over the ```Findlib``` library you should add the attribute.

Reply:
Are you sure ? I thought the `build` attribute is when you use the tool only for building, and not as a dependency, so you don't need to rebuild if the tool's version changes.

Reply:
Oops, yes, the other way around: "does not rebuild".
But anyway, that's exactly your use of ocamlfind here I believe. Or did I missed something ?